### DISCUSSION TITLE
Read coordinates from MULTILINESTRING Z

### DISCUSSION THREAD

**Discussion by SeismoFelix** [geopandas/geopandas#3198] [General]
I have the following code:

> gpd.io.file.fiona.drvsupport.supported_drivers['KML'] = 'rw'
gdf = gpd.read_file('{}/doc.kml'.format(extraction_dir),driver='KML')
print(gdf)
print(gdf.Name)
print(gdf.Description)
print(gdf.geometry)
print(gdf.geometry.iloc[0].coords)

Everything worked:

> Name Description                                           geometry
>  0  Line02              MULTILINESTRING Z ((-82.41971 28.06059 11.9500...
>  0    Line02
>  Name: Name, dtype: object
>  0    
>  Name: Description, dtype: object
>  0    MULTILINESTRING Z ((-82.41971 28.06059 11.9500...
>  Name: geometry, dtype: geometry

But in the last print, when I was trying to get the coordinates, this error happened

>NotImplementedError: Sub-geometries may have coordinate sequences, but multi-part geometries do not. 

I know the file contains coordinates. I opened the kml file, and I see this where the first two values of each group are the coordinates. 

![Screenshot 2024-02-27 at 7 49 43 PM](https://github.com/geopandas/geopandas/assets/18186011/f63ef09b-b0ac-4163-9f86-120749908928)

I am new in Geopandas, so I hope to get some guidance about extracting the coordinates for a MULTILINESTRING  geometry. Thank you.

**Comment by martinfleis** [comment]
You can use `shapely.get_coordinates(gdf.geometry.iloc[0])` instead.

**Reply by benjaminwheel3r** [reply]
This would also be an opportunity to use .explode() right?

**Reply by jorisvandenbossche** [reply]
Yes, but doing `.coords` on one an exploded geometry still does not give you all the coordinates of the full MultiLineString
